File: 0531-lonely-pixel-i/0531-lonely-pixel-i.cpp
class Solution {
public:
    int findLonelyPixel(vector<vector<char>>& picture) {
        int n = picture.size();
        int m = picture[0].size();

        vector<int> rows(n, 0), cols(m, 0);
        for (int i = 0; i < n; ++i) {
            for (int j = 0; j < m; ++j) {
                if (picture[i][j] == 'B') {
                    rows[i]++;
                    cols[j]++;
                }
            }
        }
        int cnt = 0;
        for (int i = 0; i < n; ++i) {
            for (int j = 0; j < m; ++j) {
                if (picture[i][j] == 'B' && rows[i] == 1 && cols[j] == 1)
                    cnt++;
            }
        }

        return cnt ;
    }
};

File: 0531-lonely-pixel-i/0531-lonely-pixel-i.cs
public class Solution {
    public int FindLonelyPixel(char[][] p) {
        int n = p.Length, m = p[0].Length;
        var r = new int[n];
        var c = new int[m];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (p[i][j] == 'B') {
                    r[i]++;
                    c[j]++;
                }
            }
        }
        var cnt = 0;
        for (int i = 0; i < n; ++i) {
            for (int j = 0; j < m; ++j) {
                if (p[i][j] == 'B' && r[i] == 1 && c[j] == 1)
                    cnt++;
            }
        }
        return cnt;
    }
}